src/interpreter/ByteCode.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2022-present Samsung Electronics Co., Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Walrus.h"
+
+#include "interpreter/ByteCode.h"
+
+namespace Walrus {
+
+// Return static size of opcodes
+size_t ByteCode::getBaseSize(OpcodeKind opcode)
+{
+    ASSERT(g_byteCodeInfo[opcode].m_size > 0);
+
+    return g_byteCodeInfo[opcode].m_size;
+}
+
+// Returns real size of generated opcodes
+size_t ByteCode::getSize(FunctionType* currentFunction, Module* module)
+{
+    OpcodeKind opcode = this->opcode();
+
+    ASSERT(g_byteCodeInfo[opcode].m_size > 0);
+
+    switch (opcode) {
+    case EndOpcode: {
+        return g_byteCodeInfo[EndOpcode].m_size + sizeof(ByteCodeStackOffset) * currentFunction->result().size();
+    }
+    case ThrowOpcode: {
+        Throw* throwCode = static_cast<Throw*>(this);
+
+        ASSERT(module->numberOfFunctions() >= module->numberOfTagTypes());
+
+        size_t resultSize = sizeof(ByteCodeStackOffset) * module->function(module->tagType(throwCode->tagIndex())->sigIndex())->functionType()->result().size();
+
+        return g_byteCodeInfo[ThrowOpcode].m_size + resultSize;
+    }
+    case BrTableOpcode: {
+        return g_byteCodeInfo[BrTableOpcode].m_size + sizeof(int32_t) * static_cast<BrTable*>(this)->tableSize();
+    }
+    case CallOpcode: {
+        Call* call = static_cast<Call*>(this);
+
+        ASSERT(module->numberOfFunctions() > call->index());
+
+        size_t paramSize = sizeof(ByteCodeStackOffset) * module->function(call->index())->functionType()->param().size();
+        size_t resultSize = sizeof(ByteCodeStackOffset) * module->function(call->index())->functionType()->result().size();
```

Comment:
`module->function(call->index())->functionType()` could be moved to a local variable to reduce the code.
`sizeof(ByteCodeStackOffset) * (functionType->param().size() + functionType->result().size())`

Reply:
I have moved these to local variables.